_troubleshooting/timescaledb/cagg-watermark-in-future.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Continuous aggregate watermark is in the future
+section: troubleshooting
+topics: [continuous aggregates]
+apis:
+  - [continuous aggregates, CREATE MATERIALIZED VIEW (Continuous Aggregate)]
+  - [continuous aggregates, add_continuous_aggregate()]
+  - [continuous aggregates, refresh_continuous_aggregate()]
+keywords: [continuous aggregates, real-time aggregates]
+tags: [continuous aggregates, query]
+---
+
+Continuous aggregates use a watermark to indicate which time buckets have
+already been materialized. When you query a continuous aggregate, your query
+returns materialized data from before the watermark. It returns real-time,
+non-materialized data from after the watermark.
+
+In certain cases, the watermark might be in the future. If this happens, all
```

Comment:
Technically, it doesn't need to be in the future. It could also be at a timestamp that is newer than the data being ingested, albeit in the past relative to `now()`. I understand that this will confuse users though.

Reply:
Is that scenario related to this? https://docs.timescale.com/timescaledb/latest/how-to-guides/continuous-aggregates/troubleshooting/#updates-to-previously-materialized-regions-are-not-shown-in-real-time-aggregates

We had previously documented the situation of trying to insert historical data and how that doesn't work with real-time aggregates. I think it does make sense to keep this and the future situation separate, though, because as you pointed out, it is mega confusing. Also this case is probably "highly undesired" while the other might just be "we can probably live with this trade-off"